massa-wallet/src/lib.rs

```diff
@@ -5,46 +5,73 @@
 #![warn(unused_crate_dependencies)]
 
 pub use error::WalletError;
+
+use aes_gcm_siv::aead::{Aead, NewAead};
+use aes_gcm_siv::{Aes256GcmSiv, Key, Nonce};
 use massa_hash::Hash;
 use massa_models::address::Address;
 use massa_models::composite::PubkeySig;
 use massa_models::prehash::{Map, Set};
 use massa_models::signed::Signed;
 use massa_models::{Operation, SignedOperation};
-use massa_signature::{derive_public_key, sign, PrivateKey, PublicKey};
+use massa_signature::{
+    derive_public_key, generate_random_private_key, sign, PrivateKey, PublicKey,
+};
 use serde::{Deserialize, Serialize};
 use std::path::PathBuf;
 
 mod error;
 
+/// Nonce slice used for cypher
+/// TODO: change this? not sure if we must
+pub const NONCE_SLICE: &[u8] = b"unique nonce";
```

Comment:
This will break the security of the encryption (see documentation)

* the nonce should be a random 96 bits (12 bytes) without any structure (random bytes)
* the nonce should never be reused (hence the name: N ONCE = number used once)

This means that:
* a new 12-byte random nonce should be generated every time the wallet is encrypted
* the generated nonce should be saved inside the wallet file on disk so that it can then be loaded for decryption

Here is an example: https://github.com/RCasatta/firma/blob/master/lib/src/offline/decrypt.rs#L51-L84

Reply:
Done 👍